### PR TITLE
AI Assistant: Support backend prompt building when composing forms

### LINKED ISSUE #32613: AI Extension: Move Form prompt to the backend
Let's start the process of moving prompts to the backend with the Form, that has only one possibility and is easier to transfer and test.

### PR DESCRIPTION
**Pull request by lhkowalski**
Fixes #32613.

Depends on D119078-code.

## Proposed changes:

* Change messages type to turn the `content` field optional and to add the `context` property
* Start sending the `context` on form completion requests

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Apply D119078-code to your sandbox and sandbox the `public-api` domain
* Go to the block editor and open the browser development console
* Insert a Jetpack Form block
* Ask for some form on the AI Extension (for example, a "burguer ordering form")
* Notice the prompt structure on the request debug logs:
   * Before this change, the full prompt for the form creation would be present on the request
   * After this change, we are adding a `context` property with the necessary context to build the prompt on the server side; the original prompt is still being sent so we do a swift transition

Screenshot 2023-08-21 at 16 43 32

* Make sure the form is generated correctly
* Make sure followup requests are handled properly as well

### CODE REVIEW

**Review by retrofox — COMMENTED** [review]
Can we keep sending the content? And handle this case in the endpoint? So the switching process will be smooth.

**Review by dhasilva — APPROVED** [review]
LGTM. Well done!

**Comment by dhasilva** [comment]
[DO NOT MERGE] - The diff needs to be landed first. Once that is done we can merge this one

File: projects/plugins/jetpack/extensions/blocks/ai-assistant/lib/prompt/index.ts
Comment (retrofox) [inline projects/plugins/jetpack/extensions/blocks/ai-assistant/lib/prompt/index.ts:256]: We should use a TS type for this and another context type. Please don't take it as a blocker. Let's do it in a follow-up.

Comment (lhkowalski) [reply projects/plugins/jetpack/extensions/blocks/ai-assistant/lib/prompt/index.ts:256]: Sure! The next step involves moving more complex prompts, so I am planning to type everything since the context property will be around a lot of places. Thank you for the suggestion!

**Pull request merged**
